### PR TITLE
[mp] Document subdirectory functionality and change abstractions page sturcture

### PR DESCRIPTION
**Pull request by mattppal**
# Description
Add documentation for how to structure blocks via subdirectories and break out abstractions page as its own entity.

image

image

# How Has This Been Tested?

- [x] `mintlify dev`
- [x] `mintlify broken-links`

# Checklist
- [x] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: docs/design/core-abstractions.mdx [hunk 1]
```diff
@@ -13,127 +12,23 @@ description: "These are the fundamental concepts that Mage uses to operate."
   />
 </Frame>
 
-## Project
```
Comment (wangxiaoyou1993) [inline docs/design/core-abstractions.mdx:16]: Why doe we remove the "Project" abstraction here?

File: docs/design/core-abstractions.mdx [hunk 2]
```diff
@@ -13,127 +12,23 @@ description: "These are the fundamental concepts that Mage uses to operate."
...
 
-Each pipeline is represented by a YAML file. Here is an [example](https://github.com/mage-ai/demo%5Fetl%5Fpipeline/blob/master/pipelines/etl%5Fdemo/metadata.yaml).
-
-This is what it could look like in the notebook UI:
+Each pipeline is represented by a YAML file (here's an [example](https://github.com/mage-ai/demo%5Fetl%5Fpipeline/blob/master/pipelines/etl%5Fdemo/metadata.yaml)) in the `pipelines` folder under the Mage project directory.
 
 <img
   alt="Pipeline"
-  src="https://github.com/mage-ai/mage-ai/raw/master/media/data-pipeline-overview.jpg"
+  src="media/overview/mage-build.gif"
```
Comment (wangxiaoyou1993) [inline docs/design/core-abstractions.mdx:23]: Does the relative path work?

File: docs/design/abstractions/project-structure.mdx [hunk 3]
```diff
@@ -0,0 +1,115 @@
+---
+title: "Mage file directory"
+description: "Details about how the Mage file directory works and how typical projects are structured."
+---
+
+## Overview
+
+A Mage workspace lives inside it's own folder. It's there where mage data is stored (`mage_data`) and individual projects are housed. You can think of a project as an environment: it contains configurations, pipelines, and other project-specific data. A workspace can hold multiple projects.
```
Comment (wangxiaoyou1993) [inline docs/design/abstractions/project-structure.mdx:8]: The definition might not be proper. It's conflicted with the workspace definition in this doc: https://docs.mage.ai/production/developing-in-the-cloud/workspaces/overview

File: docs/design/core-abstractions.mdx (hunk 2 above)
Comment (mattppal) [reply docs/design/core-abstractions.mdx:29]: yes

File: docs/design/abstractions/project-structure.mdx [hunk 4]
```diff
@@ -0,0 +1,114 @@
+---
+title: "Project Structure"
+description: "Details about how the Mage file directory works and how typical projects are structured."
+---
+
+## Overview
+
+A Mage project lives inside it's own folder. It's there where mage data is stored (`mage_data`) and individual projects are housed. You can think of a project as an environment: it contains configurations, pipelines, and other project-specific data. A [workspace](production/developing-in-the-cloud/workspaces/overview) can hold multiple projects.
```
Comment (wangxiaoyou1993) [inline docs/design/abstractions/project-structure.mdx:8]: 1. The link with relative path doesn't work.
2. A workspace is for one project. That doc is about managing multiple workspaces.

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**